client-app/src/pages/account/views/account-payments/account-payments.ts

```diff
@@ -1,60 +1,84 @@
 import { BvTableCtxObject } from "bootstrap-vue";
 import PaymentsFilter from "@account/components/payments-filter/index.vue";
-import { SET_PAYMENTS_LIST_CONFIG, FETCH_PAYMENTS } from '@account/store/modules/payments-list/definitions';
-import { PaymentsList, PaymentsListConfig, PaymentsListFilters } from '@account/store/modules/payments-list/types';
+import { SET_PAYMENTS_SEARCH_CRITERIA } from '@account/store/modules/payments-list/definitions';
+import { IPaymentSearchCriteria, IPaymentSearchResult, PaymentSearchCriteria } from '@common/api/api-clients';
 import { pageSizes } from "@common/constants";
+import { PaymentSearchQuery } from '@common/models/search/extensions/payment-search-query';
+import { QueryBuilder } from '@common/services/query-builder.service';
+import "@common/models/search/extensions/payment-search-criteria";
 
 const paymentsListModule = namespace('paymentsListModule');
 
 @Component({
   components: {
     PaymentsFilter
+  },
+  beforeRouteUpdate: function (to: Route, from: Route, next: (to?: RawLocation | false | ((vm: AccountPayments) => any) | void) => void) {
+    (this as AccountPayments).routeChanged(to);
+    next();
   }
 })
 export default class AccountPayments extends Vue {
-  @paymentsListModule.Getter("paymentsList")
-  private paymentsList!: PaymentsList;
-
   @paymentsListModule.Getter("isLoading")
   private isLoading!: boolean;
 
-  @paymentsListModule.Action(FETCH_PAYMENTS)
-  private fetchPayments!: () => PaymentsList;
+  @paymentsListModule.Getter("columns")
+  private columns!: boolean;
+
+  @paymentsListModule.Getter("searchCriteria")
+  private searchCriteria!: IPaymentSearchCriteria;
+
+  @paymentsListModule.Action(SET_PAYMENTS_SEARCH_CRITERIA)
+  private setSearchCriteria!: (searchCriteria: IPaymentSearchCriteria) => void;
 
-  @paymentsListModule.Action(SET_PAYMENTS_LIST_CONFIG)
-  private setListConfig!: (listConfig: PaymentsListConfig) => void;
+  @paymentsListModule.Getter("payments")
+  private payments!: IPaymentSearchResult;
 
   pageSizes = pageSizes;
 
+  queryBuilder = new QueryBuilder(PaymentSearchCriteria, PaymentSearchQuery);
+
   mounted() {
-    this.fetchPayments();
+    this.routeChanged(this.$route);
+  }
+
+  routeChanged(route: Route) {
+    const searchCriteria = this.queryBuilder.parseQuery(route.query);
+    this.setSearchCriteria({
+      ...this.searchCriteria,
+      ...searchCriteria
+    });
   }
 
-  pageChanged(page: number) {
-    this.setListConfig({ ...this.paymentsList.listConfig, pageNumber: page });
+  pageChanged(pageNumber: number) {
+    this.searchCriteriaChanged({ ...this.searchCriteria, pageNumber });
   }
 
   pageSizeChanged(pageSize: number) {
-    this.setListConfig({ ...this.paymentsList.listConfig, pageNumber: 1, pageSize: pageSize });
+    this.searchCriteriaChanged({ ...this.searchCriteria, pageNumber: 1, pageSize });
   }
 
   sortChanged(ctx: BvTableCtxObject) {
     const sortDirection = ctx.sortDesc ? "desc" : "asc";
```

Comment:
1. could we move "desc", "asc" to dedicated constants? Maybe some sorting options.
2. these options are probably already implemented somewhere

Reply:
1. Fixed
2. Don't understand what you mean. If you mean something like enumerable with desc and asc values then no, because it auto-generated code and it has sorting as string.